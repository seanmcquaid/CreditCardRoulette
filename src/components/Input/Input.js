import React from "react";
import styled from "styled-components/native";
import { TextInput, Alert } from "react-native";
import PropTypes from "prop-types";

const Input = ({onChange, placeholder, value}) => <StyledTextInput testID="textInput" onChange={onChange} placeholder={placeholder} value={value}/>


const StyledTextInput = styled(TextInput)`
    border : 2px solid black;
    padding : 10px;
    margin : 10px;
    width : 180px;
    background-color : white;
`;

Input.propTypes = {
    onChange : PropTypes.func.isRequired,
    placeholder : PropTypes.string.isRequired,
    value : PropTypes.string.isRequired,
};

Input.defaultProps = {
    onChange : () => Alert.alert("Add an onChange prop!"),
    placeholder : "Placeholder prop here",
    value : "Add value prop"
};

export default TextInput;