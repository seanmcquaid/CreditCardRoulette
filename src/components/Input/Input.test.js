import React from "react";
import {render, fireEvent} from "react-native-testing-library";
import Input from "./Input";

describe("<TextInput/>", () => {
    it("Snapshot matches correctly", () => {
        const onChange = jest.fn();

        const props = {
            onChange,
            placeholder : "Placeholder",
            value : "Value"
        };

        const input = render(<Input {...props}/>);

        expect(input).toMatchSnapshot();
    });

    it("OnChange works correctly", () => {
        const onChange = jest.fn();

        const props = {
            onChange,
            placeholder : "Placeholder",
            value : "Value"
        };

        const {getByTestId} = render(<Input {...props}/>);

        fireEvent.changeText(getByTestId("textInput"), "Test");

        expect(onChange).toHaveBeenCalledWith("Test");
    });
});