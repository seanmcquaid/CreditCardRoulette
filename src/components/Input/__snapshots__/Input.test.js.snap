// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput/> Snapshot matches correctly 1`] = `
<TextInput
  allowFontScaling={true}
  onChange={[MockFunction]}
  placeholder="Placeholder"
  rejectResponderTermination={true}
  underlineColorAndroid="transparent"
  value="Value"
/>
`;
